uint32 id
float32 x_offset
float32 distance_estimate
bool valid
